### PR TITLE
Add extra_args to FileResult

### PR DESCRIPTION
**Pull request by ordonezf**
* Enables the use of all the parameters available for to_csv and to_excel
* Update documentation

### CODE REVIEW

**Comment by ordonezf** [comment]
[Tech Debt :: Laika :: Parametros extra a to_csv/to_excel](https://trello.com/c/CN1FUnxV/7475-tech-debt-laika-parametros-extra-a-tocsv-toexcel)

File: docs/results.rst [hunk 1]
```diff
@@ -19,16 +19,30 @@ configurations for this result are:
...
 
 Example of a file result:
 
 .. code:: json
 
     "result": {
       "type": "file",
-      "filename": "output.csv"
+      "filename": "output_{yesterday}.csv",
+      "variables": {"yesterday": "d-1d"},
```
Comment (gseva) [inline docs/results.rst]: Wouldn't add templates to this example to avoid confusion

File: docs/results.rst [hunk 2]
```diff
@@ -19,16 +19,30 @@ configurations for this result are:
 
 -  filename: path to the file. Depending on the file extension this file
    will be saved as excel (xls or xlsx), tsv or csv.
+-  encoding: Defaults to 'utf-8'.
```
Comment (gseva) [inline docs/results.rst]:
```suggestion
-  encoding: Defaults to "utf-8".
```

File: docs/results.rst [hunk 3]
```diff
@@ -19,16 +19,30 @@ configurations for this result are:
...
+-  index: Write index. True by default.
+-  header: Write column names. True by default.
+-  float_format: example_.
```
Comment (gseva) [inline docs/results.rst]: Not sure if this is a good example, it doesn't use to_csv/to_excel, and it doesn't mention old-style formatting (ex: `%d`)

File: docs/results.rst [hunk 4]
```diff
@@ -19,16 +19,30 @@ configurations for this result are:
...
 .. code:: json
 
     "result": {
       "type": "file",
-      "filename": "output.csv"
+      "filename": "output_{yesterday}.csv",
+      "variables": {"yesterday": "d-1d"},
+      "index": false,
+      "header": false,
+      "extra_args": {"sheet_name": "Very Important Sheet", "start_row": 3}
```
Comment (gseva) [inline docs/results.rst]: Would use arguments that can actually apply to csv (as the example uses csv). I.e. sep, quotechar, etc.

File: docs/results.rst [hunk 5]
```diff
@@ -19,16 +19,30 @@ configurations for this result are:
 
 -  filename: path to the file. Depending on the file extension this file
    will be saved as excel (xls or xlsx), tsv or csv.
+-  encoding: Defaults to 'utf-8'.
+-  index: Write index. True by default.
+-  header: Write column names. True by default.
```
Comment (gseva) [inline docs/results.rst]: `True` -> `true` in both cases

File: laika/reports.py [hunk 6]
```diff
@@ -862,8 +862,8 @@ def write_data(self, path_or_buf):
         rest as csv.
         """
         args = dict(encoding=self.encoding, index=self.index,
-                    float_format=self.float_format, header=self.header,
-                    **self.extra_args)
+                    float_format=self.float_format, header=self.header)
+        args.update(**self.extra_args)
```
Comment (gseva) [inline laika/reports.py]: no need to unpack in this case, `args.update(self.extra_args)` is just fine

**Pull request merged**